src/Lidar.cc

```diff
@@ -192,6 +192,26 @@ bool Lidar::Update(const ignition::common::Time &/*_now*/)
   return false;
 }
 
+//////////////////////////////////////////////////
+void Lidar::ApplyNoise()
+{
+  if (this->dataPtr->noises.find(LIDAR_NOISE) != this->dataPtr->noises.end())
+  {
+    for (unsigned int j = 0; j < this->VerticalRangeCount(); ++j)
+    {
+      for (unsigned int i = 0; i < this->RangeCount(); ++i)
+      {
+        int index = j * this->RangeCount() + i;
```

Comment:
when testing with X1 Config 3 in subt, I noticed the horizontal resolution is set to `0.1`, so `RangeCount()` = 1000 samples * 0.1 res = 100 ranges. Since we're applying noise to the raw lidar buffer, we'll need to use `RayCount()` (and also `VerticalRayCount()`) otherwise noise will only be applied to the first 100 rays. 

this makes me think that there could be an existing bug for populating the laser msg [here](https://github.com/ignitionrobotics/ign-sensors/blob/63ffc47bb79c9913b5195ecc4bacb3663deba963/src/Lidar.cc#L257)..

Reply:
I think there might be multiple problems.

1. The `rendering::BaseGpuRays` class has a fixed `hResoultion` of 1. I think the `GpuLidarSensor` should set the horizontal resolution of the `BaseGpuRays`.
2. The `laserBuffer` in `GpuLidarSensor` should be sized to `RangeCount` and `VerticalRangeCount` instead of using the `RayCount`.

@iche033 , do both of those sound correct? If so, I can make those changes and then update the subt models to have a resolution of 1.

Reply:
> The rendering::BaseGpuRays class has a fixed hResoultion of 1. I think the GpuLidarSensor should set the horizontal resolution of the BaseGpuRays.

yes we maybe able to add a non-virtual function in ign-redering BaseGpuRays class to do this so it does not break ABI.

> The laserBuffer in GpuLidarSensor should be sized to RangeCount and VerticalRangeCount instead of using the RayCount.

I think this would work after proper support for resolution is added in ign-rendering `BaseGpuRays` (otherwise this [Copy](https://github.com/ignitionrobotics/ign-sensors/blob/ign-sensors4/src/GpuLidarSensor.cc#L249) call will be copying a BaseGpuRays buffer of a different size to GpuLidarSensor). 

>  update the subt models to have a resolution of 1.

for the case of subt, it looks like this will affect the `laserMsg` size (currently set using `Lidar::RangeCount()`) that takes into account of resolution) but `pointMsg` should be unaffected (set using `BaseGpuRays::RangeCount()`) and ignores resolution)  

Let me know if I can help with this or updating the robot models.

Reply:
Turns out modifying the `ignition:rendering` function is quite difficult without breaking ABI. The resolution data types in the `BaseGpuRays` class are integers when they need to be double, for example. 

So, I'm making the smaller change of using `RayCount()` and `VerticalRayCount` here. Then I'll change the SubT models to all have a resolution of 1.0.

Reply:
ouch, ok I see. The workaround here sounds good.

Reply:
Could you keep the the `hResolution` as an integer, but treat it as a denominator?